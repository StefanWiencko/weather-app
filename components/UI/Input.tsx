import {
  StyleSheet,
  Text,
  TextInput,
  TextInputProps,
  View,
  ViewStyle,
} from "react-native";
import React from "react";
import { Colors } from "@/constants/colors";

type Props = {
  label: string;
  textInputConfig?: TextInputProps;
  style?: ViewStyle;
};

const Input = ({ label, textInputConfig, style }: Props) => {
  return (
    <View style={[styles.inputContainer, style]}>
      <Text style={styles.label}>{label}</Text>
      <TextInput style={styles.input} {...textInputConfig} />
    </View>
  );
};

export default Input;

const styles = StyleSheet.create({
  inputContainer: {
    marginHorizontal: 4,
    marginVertical: 8,
  },
  label: {
    fontSize: 16,
    color: Colors.font,
    marginBottom: 4,
  },
  input: {
    backgroundColor: Colors.accent200,
    padding: 6,
    borderRadius: 6,
    fontSize: 18,
    color: Colors.font,
    width: "100%",
  },
});
