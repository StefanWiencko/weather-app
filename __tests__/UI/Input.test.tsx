import Input from "@/components/UI/Input";
import { render, fireEvent } from "@testing-library/react-native";

describe("Input", () => {
  it("displays text correctly", () => {
    const { getByDisplayValue } = render(
      <Input
        label="Test"
        textInputConfig={{
          value: "Test",
          inputMode: "text",
          onChangeText: () => {},
        }}
      />
    );

    const input = getByDisplayValue("Test");
    expect(input).toBeTruthy();
  });

  it("fires onChangeText fn", () => {
    const mockOnChangeText = jest.fn();
    const { getByDisplayValue } = render(
      <Input
        label="Test"
        textInputConfig={{
          value: "Test",
          inputMode: "text",
          onChangeText: mockOnChangeText,
        }}
      />
    );

    const input = getByDisplayValue("Test");
    fireEvent.changeText(input);

    expect(mockOnChangeText).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const tree = render(
      <Input
        label="Test"
        textInputConfig={{
          value: "Test",
          inputMode: "text",
          onChangeText: () => {},
        }}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
