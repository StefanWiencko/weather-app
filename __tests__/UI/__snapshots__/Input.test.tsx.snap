// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<View
  style={
    [
      {
        "marginHorizontal": 4,
        "marginVertical": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#222",
        "fontSize": 16,
        "marginBottom": 4,
      }
    }
  >
    Test
  </Text>
  <TextInput
    inputMode="text"
    onChangeText={[Function]}
    style={
      {
        "backgroundColor": "#ccccff",
        "borderRadius": 6,
        "color": "#222",
        "fontSize": 18,
        "padding": 6,
        "width": "100%",
      }
    }
    value="Test"
  />
</View>
`;
